### PR TITLE
Add engagement media event count endpoint

### PR DESCRIPTION
**Pull request by indrekj**
https://github.com/salemove/web_services/pull/597

### CODE REVIEW

**Comment by deiwin** [comment]
Please update the commit (after ensuring no changes are staged, run `git commit --amend --no-edit && git push --force`). [This will re-enable the merge button](https://www.pivotaltracker.com/story/show/107022596) until there's no fixup commits in the PR.

File: apiary.apib [hunk 1]
```diff
@@ -1402,6 +1402,53 @@ Possible dimensions that can be used in filters:
     ]
 
 
+### Engagement Media Event Count [POST /engagements/media_events/stats/count]
+Returns engagement media event count.
+
+Possible dimensions that can be used in filters:
+* `site_id` - Site ID
+* `operator_id` - Operator ID
+* `media` - Media (`video`, `audio`, `phone`, `chat`)
```
Comment (deiwin) [inline apiary.apib]: are `audio` and `phone` really separate?

File: apiary.apib [hunk 2]
```diff
@@ -1402,6 +1402,53 @@ Possible dimensions that can be used in filters:
...
+        "start_date": "2015-09-01T00:00:00Z",
+        "end_date": "2015-09-03T00:00:00Z",
+        "filter": {}
+      }
+
++ Response 200 (application/json)
+
+    [
+        {
+            "seconds": 1334,
```
Comment (deiwin) [inline apiary.apib]: This is not a count

File: apiary.apib [hunk 3]
```diff
@@ -1402,6 +1402,53 @@ Possible dimensions that can be used in filters:
     ]
 
 
+### Engagement Media Event Count [POST /engagements/media_events/stats/count]
+Returns engagement media event count.
```
Comment (deiwin) [inline apiary.apib]: Should be more descriptive. Something like `Returns the count of various media related events. Such as engagement offers with a particular media from both the operator and the visitor and upgrades from one media to another.`

Comment (deiwin) [inline apiary.apib]: `[...] **engagement** media related events.`

Comment (deiwin) [inline apiary.apib]: `Returns the count of various media related events. Such as engagement offers (from both the operator and the visitor) with a particular media and upgrades from one media to another.`

File: apiary.apib (hunk 1 above)
Comment (indrekj) [inline apiary.apib]: it is, but it's a good question. should we really track it separately. Depends if we want to display `phone` when doing `visitor_media_upgrade` event.

File: apiary.apib (hunk 2 above)
Comment (indrekj) [inline apiary.apib]: woop, how did this get here

File: apiary.apib (hunk 3 above)
Comment (indrekj) [inline apiary.apib]: Thank you

File: apiary.apib (hunk 1 above)
Comment (indrekj) [inline apiary.apib]: Actually it makes sense. e.g. we are displaying a chart which says if operator started a call using audio or phone. So it has a real use case.

screen shot 2015-11-09 at 22 38 10

**Comment by deiwin** [comment]
+1

**Comment by madis** [comment]
!squash

**Pull request merged**